added glGetDrawableSize to SDL.Video.OpenGL

schell opened the pull request:
To properly use glViewport on retina macs I needed to expose glGetDrawableSize.

polarina commented:
What is this `stack.yaml` file?

schell commented:
`stack.yaml` is a configuration file for the [`stack` command](http://docs.haskellstack.org/en/stable/README/), which is a nifty build tool. It's a bit like `cabal` but one level higher - it uses cabal to resolve and build your packages using a curated subset of hackage called [stackage](https://www.stackage.org/).  

I switched to `stack` from cabal sandboxes this past week and so far it's been great.

ekmett commented:
I've taken to not checking in stack.yaml, but rather checking in stack-7.8.yaml, stack-7.10.yaml, etc. and setting up things like travis to explicitly invoke them. This way if the user wants to use a custom stack.yaml config it doesn't conflict with the repository contents. I'm not yet consistent about this across all packages, but I'm headed that way.

schell commented:
By all means, it's not a requirement in any way - it was actually an accident. Feel free to move or delete it - or ask me to and I'll make another commit.

schell commented:
Any movement here?

ocharles commented:
Sorry schell! Will try and put some time aside soon and catch up on issues and PRs.

schell commented:
Thanks @ocharles :+1:

schell commented:
Any luck @ocharles?

schell commented:
Alternatively @ocharles you could add me as a contributer and I can try to merge some of these PRs - I don't have the history on the project but I have a bit of time every night.

schell commented:
@ekmett  I've removed the stack.yaml
I also resolved a conflict in .gitignore and rebased with the current changes in master since this PR was originally submitted.

tejon commented:
Hey @schell! I was added as a maintainer just last week, since I've got slightly more free time... I've gathered up the trivial stuff into the 2.1.2 release, but haven't yet gotten to the PRs that require actual reading and thinking. (I've also updated the master branch `.gitignore`. It now includes `stack.yaml`, so that should no longer be an issue.)

Could you post a small test-case example demonstrating usage and output for `glGetDrawableSize`? That's the main thing that still needs to happen, here. We should probably have tests in general, heh...

schell commented:
Oh hi @tejon - good to hear you're on the team 👍 . 

I'm using `glGetDrawableSize` in my pet project (https://github.com/schell/gelatin/blob/master/gelatin-sdl2/src/Gelatin/SDL2.hs#L52). It basically returns the framebuffer's width and height, which is different from the window's width and height on retina screens. I'm not sure how you would write a test for it though, since it's dependent on the monitor's resolution and scaling.

tejon commented:
We only need to test the wrapper, not the SDL function. :) No need for exhaustive checking, just verification of expected output from a known case or two.

schell commented:
@tejon - I can't find any test suite, how should I go about testing the wrapper? Add it to any example executable of my choosing?

tejon commented:
Yeah, until we actually build a test suite I think that's fine. As long as it can be seen in action.

tejon commented:
Made a bit of time to test this out myself, all is well. I'm merging it to a new branch, `master-next`, because IMO current `master` ought to match what's on Hackage.
